odoo/addons/base/models/res_company.py

```diff
@@ -294,7 +294,7 @@ def write(self, values):
             # and thus needs to invalidate the assets cache when this is changed
             self.env.registry.clear_cache('assets')  # not 100% it is useful a test is missing if it is the case
 
-        if 'parent_id' in values:
+        if 'parent_id' in values and any((company.exists() and company.parent_id) for company in self):
```

Comment:
the  .exists() doesn't seem necessary (this is adding a query (to check if the record exist in database) and I'm not sure which case this would solve). the added tests seems to work ok with

```suggestion
        if 'parent_id' in values and any(company.parent_id for company in self):
```

Reply:
Yes I do agree...but I added `self.exists()` because it's allowed to write translations for non-existing companies as this test shows https://github.com/odoo/odoo/blob/66a94b27ed6bc82b60708948a9e579a468af5153/odoo/addons/base/tests/test_translate.py#L1275-L1285

Reply:
if I do this change:

```py
-        if 'parent_id' in values and any((company.exists() and company.parent_id) for company in self):
+        if 'parent_id' in values and any(company.parent_id for company in self):
```
the test `test_write_non_existing` pass before and after

I have no idea what this test_write_non_existing is meant for, but in general unless there is a use case that justify it, if there is an error because we write on a record that doesn't exist, the error should not be ignored (otherwise we would need .exists at nearly each line of odoo)